src/backend/distributed/utils/maintenanced.c

```diff
@@ -253,7 +253,7 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 	/*
 	 * Look up this worker's configuration.
 	 */
-	LWLockAcquire(&MaintenanceDaemonControl->lock, LW_SHARED);
+	LWLockAcquire(&MaintenanceDaemonControl->lock, LW_EXCLUSIVE);
```

Comment:
This is the only place we use `LW_SHARED`, so it doesn't actually change anything. But since we modify `workerPid` while holding this lock, `LW_EXCLUSIVE` seems best